router-tests/integration_test.go

```diff
@@ -365,6 +367,25 @@ func prettifyJSON(t *testing.T, jsonStr string) string {
 	return res.String()
 }
 
+func TestQueryWithLogging(t *testing.T) {
```

Comment:
There are defaults logs that are printed when starting the router in INFO level. It would be a great test case to cover.